src/Build.UnitTests/Graph/GetCompatiblePlatformGraph_Tests.cs

```diff
@@ -0,0 +1,311 @@
+                                                    <Project>
+                                                    <PropertyGroup>
+                                                        <PlatformLookupTable>x86=;b=d</PlatformLookupTable>
+                                                        <Platforms>x64;x86</Platforms>
+                                                    </PropertyGroup>
+                                                    </Project>");
+
+                IDictionary<string, string> globals = new Dictionary<string, string>();
+                globals.Add("Platform", "AnyCPU");
+                ProjectGraph graph = new ProjectGraph(entryProject.Path, globalProperties: globals);
+                GetFirstNodeWithProjectNumber(graph, 2).ProjectInstance.GlobalProperties["Platform"].ShouldBe("x86");
+            }
+        }
+
+        [Fact]
+        public void PlatformIsChosenAsDefault()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                // PlatformLookupTable always takes priority. It is typically user-defined.
+                TransientTestFile entryProject = CreateProjectFile(env, 1, extraContent: @"<PropertyGroup>
+                                                                                                <EnableDynamicPlatformResolution>true</EnableDynamicPlatformResolution>
+                                                                                                <Platform>x64</Platform>
+                                                                                            </PropertyGroup>
+                                                                                            <ItemGroup>
+                                                                                                <ProjectReference Include=""$(MSBuildThisFileDirectory)2.proj"" >
+                                                                                                </ProjectReference>
+                                                                                            </ItemGroup>");
+                var file = env.CreateFile("2.proj", @"
+                                                    <Project>
+                                                    <PropertyGroup>
+                                                        <platform>x64</platform>
+                                                        <Platforms>x86;AnyCPU</Platforms>
+                                                    </PropertyGroup>
+                                                    </Project>");
+                IDictionary<string, string> globals = new Dictionary<string, string>();
+                globals.Add("Platform", "x64");
+                ProjectGraph graph = new ProjectGraph(entryProject.Path, globalProperties: globals);
+                GetFirstNodeWithProjectNumber(graph, 2).ProjectInstance.GlobalProperties["Platform"].ShouldBe(String.Empty);
```

Comment:
you mean the evaluated property should be x64 right? not the global property.

Reply:
Scratch this comment. 2.proj's global prop for `platform` _should_ be empty because it'll evaluate to `x64` anyway. Can we add a check to verify that its evaluated `Platform` matches 1's platform?

Reply:
added in